lib/dependencies/AMDDefineDependency.js

```diff
@@ -2,113 +2,137 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var NullDependency = require("./NullDependency");
+"use strict";
+const NullDependency = require("./NullDependency");
 
-function AMDDefineDependency(range, arrayRange, functionRange, objectRange) {
-	NullDependency.call(this);
-	this.range = range;
-	this.arrayRange = arrayRange;
-	this.functionRange = functionRange;
-	this.objectRange = objectRange;
+class AMDDefineDependency extends NullDependency {
+	constructor(range, arrayRange, functionRange, objectRange) {
+		super();
+		this.range = range;
+		this.arrayRange = arrayRange;
+		this.functionRange = functionRange;
+		this.objectRange = objectRange;
+	}
+
+	get type() {
+		return "amd define";
+	}
 }
-module.exports = AMDDefineDependency;
 
-AMDDefineDependency.prototype = Object.create(NullDependency.prototype);
-AMDDefineDependency.prototype.constructor = AMDDefineDependency;
-AMDDefineDependency.prototype.type = "amd define";
+AMDDefineDependency.Template = class AMDDefineDependencyTemplate {
+	get definitions() {
+		return {
+			f: [
+				"var __WEBPACK_AMD_DEFINE_RESULT__;",
+				`!(__WEBPACK_AMD_DEFINE_RESULT__ = #.call(exports, __webpack_require__, exports, module),
+				__WEBPACK_AMD_DEFINE_RESULT__ !== undefined && (module.exports = __WEBPACK_AMD_DEFINE_RESULT__))`
+			],
+			o: [
+				"",
+				"!(module.exports = #)"
+			],
+			of: [
+				"var __WEBPACK_AMD_DEFINE_FACTORY__, __WEBPACK_AMD_DEFINE_RESULT__;",
+				`!(__WEBPACK_AMD_DEFINE_FACTORY__ = (#),
+				__WEBPACK_AMD_DEFINE_RESULT__ = (typeof __WEBPACK_AMD_DEFINE_FACTORY__ === 'function' ?
```

Comment:
This is adding extra indentation that wasn't there before in the output.

It's probably not a problem, but it could be worth it to use `common-tags`' `stripIndent`:

```js
const stripIndent = require("common-tags/lib/stripIndent");

/* ... */

				stripIndent`!(__WEBPACK_AMD_DEFINE_FACTORY__ = (#),
				__WEBPACK_AMD_DEFINE_RESULT__ = (typeof __WEBPACK_AMD_DEFINE_FACTORY === 'function' ?
				${/* ... */""}`
```